commit c05a8c8bf401397e9b888bdf5d749ca07770d54d
Author: Gabriel Krisman Bertazi <krisman@linux.vnet.ibm.com>
Date:   Fri Mar 17 00:48:32 2017 +0000

    serial: 8250_pci: Detach low-level driver during PCI error recovery
    
    [ Upstream commit f209fa03fc9d131b3108c2e4936181eabab87416 ]
    
    During a PCI error recovery, like the ones provoked by EEH in the ppc64
    platform, all IO to the device must be blocked while the recovery is
    completed.  Current 8250_pci implementation only suspends the port
    instead of detaching it, which doesn't prevent incoming accesses like
    TIOCMGET and TIOCMSET calls from reaching the device.  Those end up
    racing with the EEH recovery, crashing it.  Similar races were also
    observed when opening the device and when shutting it down during
    recovery.
    
    This patch implements a more robust IO blockage for the 8250_pci
    recovery by unregistering the port at the beginning of the procedure and
    re-adding it afterwards.  Since the port is detached from the uart
    layer, we can be sure that no request will make through to the device
    during recovery.  This is similar to the solution used by the JSM serial
    driver.
    
    I thank Peter Hurley <peter@hurleysoftware.com> for valuable input on
    this one over one year ago.
    
    Signed-off-by: Gabriel Krisman Bertazi <krisman@linux.vnet.ibm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1449_c05a8c8bf401397e9b888bdf5d749ca07770d54d.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1449_c05a8c8bf401397e9b888bdf5d749ca07770d54d.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
